4.preprocess_features/scripts/1.single_cell_processing.py

```diff
@@ -0,0 +1,173 @@
+
+# path for platemap directory
+platemap_dir = pathlib.Path("../0.download_data/metadata")
+
+# operations to perform for feature selection
+feature_select_ops = [
+    "variance_threshold",
+    "correlation_threshold",
+    "blocklist",
+    "drop_na_columns"
+]
+
+
+# ## Set dictionary with plates to process
+
+# In[3]:
+
+
+# create plate info dictionary 
+plate_info_dictionary = {
+    name: {
+        "profile_paths": {
+            "sc": str(
+                pathlib.Path(list(profiles_dir.rglob(f"{name}_sc_converted.parquet"))[0]).resolve(
+                    strict=True
+                )
+            ),
+            "organoid": str(
+                pathlib.Path(list(profiles_dir.rglob(f"{name}_organoid_converted.parquet"))[0]).resolve(
+                    strict=True
+                )
+            ),
+        },
+        "platemap_path": str(
+            pathlib.Path(list(platemap_dir.rglob(f"{name}_platemap.csv"))[0]).resolve(
+                strict=True
+            )
+        ),
+    }
+    for name in plate_names
+}
+
+# view the dictionary to assess that all info is added correctly
+pprint.pprint(plate_info_dictionary, indent=4)
+
+
+# ## Process data with pycytominer
+
+# In[4]:
+
+
+for plate, info in plate_info_dictionary.items():
+    print(f"Performing pycytominer pipeline for {plate}")
+
+    # Prepare output paths for different profile types
+    for profile in ["sc", "organoid"]:
+        output_annotated_file = str(pathlib.Path(f"{output_dir}/{plate}_{profile}_annotated.parquet"))
+        output_normalized_file = str(pathlib.Path(f"{output_dir}/{plate}_{profile}_normalized.parquet"))
+        output_feature_select_file = str(pathlib.Path(f"{output_dir}/{plate}_{profile}_feature_selected.parquet"))
+        
+        profile_path = info["profile_paths"].get(profile)
+
+        if profile_path:
+            # Read in the profile and platemap files
+            profile_df = pd.read_parquet(profile_path)
+            platemap_df = pd.read_csv(info["platemap_path"])
+
+            # Set compartments based on profile type
+            compartments = ["nuclei", "cells", "cytoplasm"] if profile == "sc" else ["organoid"]
+
+            print("Performing annotation for", plate, "for the", profile, "profiles...")
+
+            # Step 1: Annotation
+            annotate(
+                profiles=profile_df,
+                platemap=platemap_df,
+                join_on=["Metadata_well_position", "Image_Metadata_Well"],
+                output_file=output_annotated_file,
+                output_type="parquet",
+            )
+
+            # Load the annotated parquet file to fix metadata columns names
+            annotated_df = pd.read_parquet(output_annotated_file)
+
+            # Rename columns using the rename() function
+            column_name_mapping = {
+                "Image_Metadata_Site": "Metadata_Site",
+                "Image_Metadata_ZSlice": "Metadata_ZSlice"
+            }
```

Comment:
Consider defining this outside of the for loops

Reply:
That works for me, added!